zap/src/main/java/org/zaproxy/zap/eventBus/SimpleEventBus.java

```diff
@@ -284,11 +255,10 @@ public String getPublisherName() {
 
     private static class RegisteredPublisher {
         private EventPublisher publisher;
-        private String[] eventTypes;
+        private List<String> eventTypes;
         private List<RegisteredConsumer> consumers = new CopyOnWriteArrayList<>();
```

Comment:
one doubt i have, do we need CopyOnWriteArrayList here ?

Reply:
Yes, #4827.

Reply:
Marking resolved, evidence for necessity provided.